libraries/AP_BoardConfig/AP_BoardConfig.cpp

```diff
@@ -411,7 +411,7 @@ void AP_BoardConfig::throw_error(const char *err_type, const char *fmt, va_list
         uint32_t now = AP_HAL::millis();
         if (now - last_print_ms >= 5000) {
             last_print_ms = now;
-            char printfmt[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+2];
+            char printfmt[52];  // MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+2
```

Comment:
why this one ? we don't remove the include anyway, so that should be fine to use this constant ?

Reply:
What I'm playing around with here is the fact that we use MAVLink constants - and some structures - throughout our code even when `HAL_GCS_ENABLED` is false.

Without the headers you can't compile the GPS library as we use the MAVLink lock levels.

We use the mavlink message length in various places for no particular reason except that it's our usual tansport (as the patch this comment is associated with shows).

The RangeFinder library interface has mavlink rotations in it.

For AP_Periph this isn't a huge problem as we build the mavlink headers (even 'though the vast majority of periphs don't do the GCS thing!).

It is a problem as soon as you cross one of these headers with another library on either of the IOMCU or Bootloader builds, when you want to add functionality and re-use libraries.

In this PR I'm now playing with the concept of having an `AP_MAVLINK_ENABLED` - that would leave you with `HAL_GCS_ENABLED` which would give you a GCS object to do the statustexts in, but you wouldn't get `GCS_MAVLINK` backends  unless `AP_MAVLINK_ENABLED` was set.  This is to try to resolve the RC Telemetry problem where statustexts are fetched from the GCS library and that's all based around `mavlink_statustext_t`